openedx_events/event_bus/avro/serializer.py

```diff
@@ -0,0 +1,78 @@
+"""
+Serialize events to Avro records.
+"""
+import json
+from datetime import datetime
+
+import attr
+from opaque_keys.edx.keys import CourseKey
+
+from .custom_serializers import CourseKeyAvroSerializer, DatetimeAvroSerializer
+from .schema import schema_from_signal
+
+DEFAULT_SERIALIZERS = {
+    datetime: DatetimeAvroSerializer.serialize,
+    CourseKey: CourseKeyAvroSerializer.serialize,
+}
+
+
+def _get_non_attrs_serializer(serializers=None):
+    """Create a method to pass to attr.as_dict that will serialize using custom serializers."""
+    param_serializers = serializers or {}
+    all_serializers = {**DEFAULT_SERIALIZERS, **param_serializers}
+
+    def _serialize_non_attrs_values(inst, field, value):  # pylint: disable=unused-argument
+        for extended_class, serializer in all_serializers.items():
+            if field:
+                if issubclass(field.type, extended_class):
+                    return serializer(value)
+            if issubclass(type(value), extended_class):
+                return serializer(value)
+        return value
+    return _serialize_non_attrs_values
+
+
+def _event_data_to_record_dict(event_data, serializers=None):
```

Comment:
Let's work out the "deserializer" comments and apply here as appropriate. Then I can give this code a final look.